### PR TITLE
Revisit project versioning

### LINKED ISSUE #63: Keep past versions of OLM manifests
Follow this structure https://github.com/operator-framework/operator-registry#manifest-format

### PR DESCRIPTION
**Pull request by phoracek**
I tried to do a few fixes in our versioning and release management. That should make it easier to implement upgrades and make OLM integration cleaner.

- Drop "v" from version names, it is not allowed in CSV. With this change it we are consistent across the project and SemVer compatible.
- Keep only released manifests in sources. There is no need to keep latest manifests here, they can be generated if needed. Let's not confuse anyone with non-released manifests.
- Keep previously released manifests under the source tree. That is needed for OLM integration and future upgrades.

Fixes #63

### CODE REVIEW

**Comment by phoracek** [comment]
@SchSeba @booxter could you please take a look? Mostly just moving files and cleanup.

File: README.md [hunk 1]
```diff
@@ -151,16 +151,16 @@ spec:
 First install the operator itself:
 
 ```shell
-kubectl apply -f https://raw.githubusercontent.com/kubevirt/cluster-network-addons-operator/master/deploy/namespace.yaml
-kubectl apply -f https://raw.githubusercontent.com/kubevirt/cluster-network-addons-operator/master/deploy/crds/network-addons-config.crd.yaml
-kubectl apply -f https://raw.githubusercontent.com/kubevirt/cluster-network-addons-operator/master/deploy/operator.yaml
+kubectl apply -f https://github.com/kubevirt/cluster-network-addons-operator/blob/master/manifests/cluster-network-addons/0.0.0/namespace.yaml
+kubectl apply -f https://github.com/kubevirt/cluster-network-addons-operator/blob/master/manifests/cluster-network-addons/0.3.0/network-addons-config.crd.yaml
```
Comment (SchSeba) [inline README.md]: https://github.com/kubevirt/cluster-network-addons-operator/releases/download/v0.3.0/network-addons-config.crd.yaml

this?

File: README.md [hunk 2]
```diff
@@ -308,3 +265,55 @@ spec:
...
+
+# delete the cluster
+make cluster-down
+```
+
+# Releasing
+
+Steps to create a new release:
+
+1. Test operator, make sure it deploys all components, exposes failures in NetworkAddonsConfig.Spec field as well as progressing status of components and "Ready".
```
Comment (SchSeba) [inline README.md]: NetworkAddonsConfig.Spec -> NetworkAddonsConfig.Status ?

File: manifests/cluster-network-addons/0.3.0/cluster-network-addons-operator.v0.3.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -148,20 +148,16 @@ spec:
                   - name: MULTUS_IMAGE
                     value: docker.io/nfvpe/multus:latest
                   - name: LINUX_BRIDGE_IMAGE
-                    value: quay.io/kubevirt/cni-default-plugins:latest
+                    value: quay.io/kubevirt/cni-default-plugins
```
Comment (SchSeba) [inline manifests/cluster-network-addons/0.3.0/cluster-network-addons-operator.v0.3.0.clusterserviceversion.yaml]: why you remove the latest here?

File: Makefile [hunk 4]
```diff
@@ -1,37 +1,51 @@
...
 fmt:
 	go fmt ./pkg/... ./cmd/...
 
-docker-build:
-	docker build -f build/Dockerfile -t $(IMAGE_REGISTRY)/$(OPERATOR_IMAGE):$(IMAGE_TAG) .
+docker-build-operator:
+	docker build -f build/operator/Dockerfile -t $(IMAGE_REGISTRY)/$(OPERATOR_IMAGE):$(IMAGE_TAG) .
+
+docker-build-registry:
+	docker build -f build/operator/Dockerfile -t $(IMAGE_REGISTRY)/$(REGISTRY_IMAGE):$(IMAGE_TAG) .
```
Comment (SchSeba) [inline Makefile]: build/operator/Dockerfile -> build/registry/Dockerfile ?

Comment (phoracek) [reply Makefile]: nice! thanks

File: README.md (hunk 1 above)
Comment (phoracek) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (phoracek) [reply README.md]: Done

**Comment by phoracek** [comment]
@SchSeba fixed, could you please take another look?

**Comment by phoracek** [comment]
/hold cancel

Additional commit with travis fixes

File: README.md [hunk 5]
```diff
@@ -308,3 +265,55 @@ spec:
...
+make fmt
+
+# generate manifests
+make generate-manifests
+
+# generate sources (requires operator-sdk installed on your host)
+operator-sdk generate k8s
+
+# build images (uses multi-stage builds and therefore requires Docker >= 17.05)
+make docker-build-operator
```
Comment (SchSeba) [inline README.md]: refer the new docker-build and docker-push here please

**Pull request merged**